WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -134,12 +134,13 @@ class NewGutenbergViewController: UIViewController, PostEditor, PublishingEditor
         let networkClient = NewGutenbergNetworkClient(blog: post.blog)
 
         let selfHostedApiUrl = post.blog.url(withPath: "wp-json/")
-        let siteApiRoot = post.blog.isAccessibleThroughWPCom() ? post.blog.wordPressComRestApi()?.baseURL.absoluteString : selfHostedApiUrl
+        let siteApiRoot = post.blog.isAccessibleThroughWPCom() && post.blog.isHostedAtWPcom ? post.blog.wordPressComRestApi()?.baseURL.absoluteString : selfHostedApiUrl
         let siteId = post.blog.dotComID?.stringValue
         let authToken = post.blog.authToken ?? ""
         var authHeader = "Bearer \(authToken)"
 
-        let applicationPassword = try? post.blog.getApplicationToken()
+        // TODO: Reinstate the application password token getter once supported
+        let applicationPassword = try? post.blog.getPassword()
```

Comment:
Hey David 👋 

It is available under an experimental flag, here's how you can test it out:

**Precondition**: You have to be already logged in with a different account, because this feature is only available from the "My Sites" picker.

- Enable the feature flag by going to: Me > App Settings > Debug > Feature Flags > Enable: `Application Password for self-hosted sites`
- Go to the "My Site tab"
- Open the "My Sites" Picker
- Tap on the (+) button
- Tap on Add self-hosted site
- Enter your site address
- Follow the process of login into your site
- Approve the creation of the application password 

Once you finish the process the site should be added, and the application password ready to use.

> Relates to #23499 (comment).
I'm going to merge that PR now, but what I asked there was more related to code structure, if that was a good approach, or if there are better ways. Let's go with what we have and iterate if needed.

Let me know if you encounter any issues!